<?xml version="1.0" encoding="UTF-8"?>
<db:article xmlns:db="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.2" xml:lang="en">
<db:info>
<db:title>&lt;QTest&gt;</db:title>
<db:productname>QtTestLib</db:productname>
<db:edition>Qt 6.9.1 Reference Documentation</db:edition>
<db:titleabbrev>Qt Test Reference Documentation</db:titleabbrev>
<db:abstract>
<db:para>Qt Test Reference Documentation.</db:para></db:abstract>
<db:pubdate>2025-06-29</db:pubdate>
<db:date>2025-06-29</db:date>
<db:authorgroup>
<db:author>
<db:orgname class="corporation">The Qt Company Ltd.</db:orgname>
</db:author>
</db:authorgroup>
</db:info>
<db:section xml:id="macro-documentation">
<db:title>Macro Documentation</db:title>
<db:section xml:id="QTEST_THROW_ON_FAIL">
<db:title>[since 6.8] QTEST_THROW_ON_FAIL</db:title>
<db:para>When defined, <db:link xlink:href="qtest.xml#QCOMPARE">QCOMPARE</db:link>()/<db:link xlink:href="qtest.xml#QVERIFY">QVERIFY</db:link>() etc always throw on failure. QTest::throwOnFail() then no longer has any effect.</db:para>
<db:para>This macro was introduced in Qt 6.8.</db:para>
</db:section>
<db:section xml:id="QTEST_THROW_ON_SKIP">
<db:title>[since 6.8] QTEST_THROW_ON_SKIP</db:title>
<db:para>When defined, <db:link xlink:href="qtest.xml#QSKIP">QSKIP</db:link>() always throws. QTest::throwOnSkip() then no longer has any effect.</db:para>
<db:para>This macro was introduced in Qt 6.8.</db:para>
</db:section>
</db:section>
</db:article>
